src/lib/analytics/segmentOneTrustIntegration/segmentOneTrustIntegration.ts

```diff
@@ -0,0 +1,46 @@
+import { data as sd } from "sharify"
+import { fetchDestinations } from "./fetchDestinations"
+import { getConsentAndLoad } from "./getConsentAndLoad"
+
+const SEGMENT_WRITE_KEY = sd.SEGMENT_WRITE_KEY
+const SEGMENT_MAPPING = "C0001" // assign OneTrust category to Segment itself.
+
+let PREVIOUS_CONSENT = ""
+let SEGMENT_DESTINATIONS = []
+
+let ONETRUST_ENABLED = false
+
+if (!sd.THIRD_PARTIES_DISABLED && !sd.EIGEN && sd.ONETRUST_SCRIPT_ID) {
+  ONETRUST_ENABLED = true
+}
+
+export function loadSegment() {
+  if (ONETRUST_ENABLED) {
+    // OneTrust is enabled, load Segment based on OneTrust consent.
+    fetchDestinations(SEGMENT_WRITE_KEY).then(async destinations => {
```

Comment:
the fetch happens on every page load/reload, so call volume to segment's api will increase.